hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/workflow/diagnostic/DiagnosticManager.java

```diff
@@ -0,0 +1,187 @@
+import com.blackducksoftware.integration.hub.detect.configuration.DetectProperty;
+import com.blackducksoftware.integration.hub.detect.workflow.bomtool.BomToolEvaluation;
+import com.blackducksoftware.integration.hub.detect.workflow.codelocation.DetectCodeLocation;
+
+public class DiagnosticManager {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final DetectConfigWrapper detectConfigWrapper;
+
+    private File outputDirectory;
+
+    private final DiagnosticReportManager diagnosticReportManager;
+    private final DiagnosticLogManager diagnosticLogManager;
+    private final DetectRunManager detectRunManager;
+    private final DiagnosticFileManager diagnosticFileManager;
+
+    private boolean isDiagnosticProtected = false;
+    private boolean isDiagnostic = false;
+
+    public DiagnosticManager(final DetectConfigWrapper detectConfigWrapper, final DiagnosticReportManager diagnosticReportManager, final DiagnosticLogManager diagnosticLogManager,
+            final DetectRunManager detectRunManager, final DiagnosticFileManager diagnosticFileManager) {
+        this.detectConfigWrapper = detectConfigWrapper;
+        this.diagnosticReportManager = diagnosticReportManager;
+        this.diagnosticLogManager = diagnosticLogManager;
+        this.detectRunManager = detectRunManager;
+        this.diagnosticFileManager = diagnosticFileManager;
+    }
+
+    public void init(final boolean isDiagnostic, final boolean isDiagnosticProtected) {
+
+        this.isDiagnostic = isDiagnostic;
+        this.isDiagnosticProtected = isDiagnosticProtected;
+
+        if (!isDiagnostic) {
+            return;
+        }
+
+        System.out.println("");
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
+        System.out.println("Diagnostic mode on. Run id " + detectRunManager.getRunId());
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
+        System.out.println("");
+
+        final File bdioDirectory = new File(detectConfigWrapper.getProperty(DetectProperty.DETECT_BDIO_OUTPUT_PATH));
+        this.outputDirectory = new File(detectConfigWrapper.getProperty(DetectProperty.DETECT_OUTPUT_PATH));
+        try {
+            diagnosticFileManager.init(outputDirectory, bdioDirectory, detectRunManager.getRunId());
+        } catch (final Exception e) {
+            logger.error("Failed to create diagnostics directory.");
+            e.printStackTrace();
+        }
+
+        logger.info("Initializing diagnostic managers.");
+        try {
+            diagnosticReportManager.init(diagnosticFileManager.getReportDirectory(), detectRunManager.getRunId());
+            diagnosticLogManager.init(diagnosticFileManager.getLogDirectory());
+        } catch (final Exception e) {
+            logger.error("Failed to initialize.");
+            e.printStackTrace();
```

Comment:
any reason why we're using printStackTrace in this class instead of providing the throwable to the logger?

Reply:
Oh that's a good point. Changing.